### DISCUSSION TITLE
Resume Functionality

### DISCUSSION THREAD

**Discussion by puzzlepeaches** [blacklanternsecurity/bbot#826] [Ideas]
**Description**

If a scan fails for whatever reason, it would be nice to be able to resume the scan similar to projectdiscovery tools. 

The feature would ideally function something like this:

```
bbot -t targets.txt -f subdomain-enum active web-basic cloud-enum email-enum deadly --resume [SCAN_NAME]
```

**Comment by TheTechromancer** [comment]
I agree this would be a nice feature to have; however due to BBOT's recursive nature, implementing a resume feature is extremely non-trivial. 

This is not currently on the roadmap, although it may be sometime in the future. Until then, pull requests are always welcome. Also, in my experience bash job control via `CTRL+Z` tends to work pretty well for pause/resume.

**Comment by Sh4d0wHunt3rX** [comment]
I was going to write same request, noticed this discussion. The most important part of the scan for me is dns mutations at the end, and sometimes because of whatever reason, bbot interrupted and I missed that part : (

**Reply by puzzlepeaches** [reply]
I have figured this out a bit. While it isn't a cut and dry resume option, take a look at the output_modules configuration options:

https://www.blacklanternsecurity.com/bbot/scanning/configuration/

If you keep the same scan name, you can at least reuse data you have collected previously as a seed. This is more than enough for our needs at this time.

**Reply by Sh4d0wHunt3rX** [reply]
Thanks for the solution 🙏 However, my point of having resume ability is to not spend time again to scan those all, for example my last scan took 24 hours, I don't want to repeat it all over again.
